warning: 'indices' may be used uninitialized

**Describe the bug**

Spotted on msys2:
```
[354/509] Building CXX object client/CMakeFiles/freeciv21-client.dir/tileset/layer_terrain.cpp.obj
D:/a/freeciv21/freeciv21/client/tileset/layer_terrain.cpp: In member function 'void freeciv::layer_terrain::fill_terrain_sprite_array(std::vector<drawn_sprite>&, const tile*, const terrain*, terrain**) const':
D:/a/freeciv21/freeciv21/client/tileset/layer_terrain.cpp:993:43: warning: 'indices' may be used uninitialized [-Wmaybe-uninitialized]
  993 |       int array_index = flattened_index<2>(info.matches_with.size(),
      |                         ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
  994 |                                            {indices[1], indices[0]}, i);
      |                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
D:/a/freeciv21/freeciv21/client/tileset/layer_terrain.cpp:959:26: note: 'indices' declared here
  959 |       std::array<int, 2> indices;
      |                          ^~~~~~~
D:/a/freeciv21/freeciv21/client/tileset/layer_terrain.cpp:993:43: warning: 'indices' may be used uninitialized [-Wmaybe-uninitialized]
  993 |       int array_index = flattened_index<2>(info.matches_with.size(),
      |                         ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
  994 |                                            {indices[1], indices[0]}, i);
      |                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
D:/a/freeciv21/freeciv21/client/tileset/layer_terrain.cpp:959:26: note: 'indices' declared here
  959 |       std::array<int, 2> indices;
      |                          ^~~~~~~
```
[Link to CI build](https://github.com/longturn/freeciv21/actions/runs/6999240963/job/19038367004?pr=2056)

--- a/client/tileset/layer_terrain.cpp
+++ b/client/tileset/layer_terrain.cpp
@@ -956,7 +956,7 @@ void layer_terrain::fill_terrain_sprite_array(
           MATCH(iso_dirs[(i + 5) % 6]), // Direction "before"
           MATCH(iso_dirs[i])};          // Direction "after"
 
-      std::array<int, 2> indices;
+      std::array<int, 2> indices = {0, 0};
       switch (info.style) {
       case MATCH_SAME:
         fc_assert_ret(info.matches_with.size() == 2);

